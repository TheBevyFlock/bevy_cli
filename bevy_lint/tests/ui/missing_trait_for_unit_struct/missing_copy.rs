#![feature(register_tool)]
#![register_tool(bevy)]
//~v NOTE: the lint level is defined here
#![deny(bevy::missing_copy)]

#[derive(Clone)]
//~| HELP: `Copy` can be automatically derived
//~v ERROR: defined a unit struct without a `Copy` implementation
pub struct IsDefaultUiCamera;

// This should not raise an ERROR, since `Copy` is derived.
#[derive(Copy, Clone)]
pub struct DeriveCopy;

// This should not raise an ERROR, since `Copy` is implemented.
#[derive(Clone)]
pub struct ImplCopy;

impl Copy for ImplCopy {}

// This should not raise an ERROR, since this is not a unit struct.
pub struct ComponentWithFields(#[allow(dead_code)] f32);

#[allow(bevy::missing_copy)]
// This should not raise an ERROR, since the lint is silenced.
pub struct AllowMissingCopy;
