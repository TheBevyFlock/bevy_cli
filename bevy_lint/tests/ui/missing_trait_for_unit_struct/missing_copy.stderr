error: defined a unit struct without a `Copy` implementation
  --> tests/ui/missing_trait_for_unit_struct/missing_copy.rs:12:1
   |
12 | pub struct IsDefaultUiCamera;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: `Component` implemented here
  --> tests/ui/missing_trait_for_unit_struct/missing_copy.rs:9:10
   |
 9 | #[derive(Component, Clone)]
   |          ^^^^^^^^^
note: the lint level is defined here
  --> tests/ui/missing_trait_for_unit_struct/missing_copy.rs:4:9
   |
 4 | #![deny(bevy::missing_copy)]
   |         ^^^^^^^^^^^^^^^^^^
   = note: this error originates in the derive macro `Component` (in Nightly builds, run with -Z macro-backtrace for more info)
help: `Copy` can be automatically derived
   |
12 + #[derive(Copy)]
13 | pub struct IsDefaultUiCamera;
   |

error: aborting due to 1 previous error

